src/app/app.component.ts

```diff
@@ -1,11 +1,33 @@
-import { Component} from '@angular/core';
+import { Component, OnInit} from '@angular/core';
+import { Router, NavigationEnd, ActivatedRoute } from '@angular/router';
+import { Title } from '@angular/platform-browser';
+import { filter, map, mergeMap} from 'rxjs/operators';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
-  constructor() {
+export class AppComponent implements OnInit {
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private titleService: Title
+  ) {}
+
+  ngOnInit() {
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      map(() => this.activatedRoute),
+      map((route) => {
+        while (route.firstChild) {
+          route = route.firstChild;
```

Comment:
Reason for all this? `mergeMap(() => this.activatedRoute.firstChild.data)` should've worked on NavigationEnd

Reply:
I'm unsure what the reasoning for all of it is. I looked up several examples of changing the title based on routes, and they all looked similar to this. But I think the while loop lets you get more specific title tags based on the children routes written in the other folders. So with your changes, going to /containers page will have a title of "Dockstore | Tools" but going to a specific tool will also have the same title instead of "Tools" becoming singular.

Reply:
The next commit does combine the two maps into one now though.